freqtrade/strategy/strategy_updater.py

```diff
@@ -0,0 +1,213 @@
+        'sell_profit_offset': 'exit_profit_offset',
+        'ignore_roi_if_buy_signal': 'ignore_roi_if_entry_signal',
+        'forcebuy_enable': 'force_entry_enable',
+    }
+
+    function_mapping = {
+        'populate_buy_trend': 'populate_entry_trend',
+        'populate_sell_trend': 'populate_exit_trend',
+        'custom_sell': 'custom_exit',
+        'check_buy_timeout': 'check_entry_timeout',
+        'check_sell_timeout': 'check_exit_timeout',
+        # '': '',
+    }
+    # order_time_in_force, order_types, unfilledtimeout
+    otif_ot_unfilledtimeout = {
+        'buy': 'entry',
+        'sell': 'exit',
+    }
+
+    # create a dictionary that maps the old column names to the new ones
+    rename_dict = {'buy': 'enter_long', 'sell': 'exit_long', 'buy_tag': 'enter_tag'}
+
+    def start(self, strategy_obj: dict) -> None:
+        """
+        Run strategy updater
+        It updates a strategy to v3 with the help of the ast-module
+        :return: None
+        """
+
+        self.cwd = os.getcwd()
+        self.strategies_backup_folder = f'{os.getcwd()}user_data/strategies_orig_updater'
+        source_file = strategy_obj['location']
+
+        # read the file
+        with open(source_file, 'r') as f:
+            old_code = f.read()
+        if not os.path.exists(self.strategies_backup_folder):
+            os.makedirs(self.strategies_backup_folder)
+
+        # backup original
+        # => currently no date after the filename,
+        # could get overridden pretty fast if this is fired twice!
+        shutil.copy(source_file, f"{self.strategies_backup_folder}/{strategy_obj['location_rel']}")
+
+        # update the code
+        new_code = strategy_updater.update_code(self, old_code,
+                                                strategy_updater.name_mapping,
+                                                strategy_updater.function_mapping,
+                                                strategy_updater.rename_dict)
+
+        # write the modified code to the destination folder
+        with open(source_file, 'w') as f:
+            f.write(new_code)
+        print(f"conversion of file {source_file} successful.")
+
+    # define the function to update the code
+    def update_code(self, code, _name_mapping, _function_mapping, _rename_dict):
+        # parse the code into an AST
+        tree = ast.parse(code)
+
+        # use the AST to update the code
+        updated_code = strategy_updater.modify_ast(
+            tree,
+            _name_mapping,
+            _function_mapping,
+            _rename_dict)
+
+        # return the modified code without executing it
+        return updated_code
+
+    # function that uses the ast module to update the code
+    def modify_ast(node, _name_mapping, _function_mapping, _rename_dict): # noqa
+        # create a visitor that will update the names and functions
+        class NameUpdater(ast.NodeTransformer):
+            def generic_visit(self, node):
```

Comment:
does this HAVE to be a nested class?

it has 2 downsides i can see so far
* it increases indentation by 8 spaces for no good reason (which automatically wraps more lines than we'd want)
* field names are now confusing 
  (what is `node` in the function below? the `node` from `modify_ast()` is no longer accessible, as it's covered by the `node` definition in this method ...)